src/Common/Session/SessionTracker.php

```diff
@@ -64,4 +64,40 @@ public static function updateSessionExpiration(): void
     {
         sqlStatementNoLog("UPDATE `session_tracker` SET `last_updated` = NOW() WHERE `uuid` = ?", [$_SESSION['session_database_uuid']]);
     }
+
+    // Function to update the throttle down function (ie. counting scripts)
+    //  Only basically used for the online demos to prevent abuse of demo farm
+    public static function updateSessionThrottleDown(): void
+    {
+        if ((getenv('THROTTLE_DOWN_WAIT_MILLISECONDS', true) ?? 0) > 0 ) {
+            sqlStatementNoLog("UPDATE `session_tracker` SET `number_scripts` = `number_scripts` + 1 WHERE `uuid` = ?", [$_SESSION['session_database_uuid']]);
+        }
+    }
+
+    // Function to throttle down requests when using the online demos to prevent abuse of the demo farm
+    public static function processSessionThrottleDown(): void
+    {
+        $throttleDownWaitMilliseconds = getenv('THROTTLE_DOWN_WAIT_MILLISECONDS', true) ?? 0;
+        if ($throttleDownWaitMilliseconds > 0 ) {
+            // calculate $timeThrottle['time_throttle'], which will be average time (in milliseconds) per script call
+            $timeThrottle = sqlQueryNoLog("SELECT `number_scripts`, `created`, NOW() as `current_timestamp` FROM `session_tracker` WHERE `uuid` = ?", $_SESSION['session_database_uuid']);
```

Comment:
made this feat look much easier than ip blocking :sunglasses: 

doesn't the bind have to be in an array for the new adodb thing that popped up the other day?

Reply:
goodness, i copied that code from another block. Odd that it is working. Will fix it here and in original code.

Reply:
willing to bet that adodb is a bit flexible on this when 1 parameter, however bet breaks if it is null or empty